src/main/scala/firrtl_interpreter/CircuitState.scala

```diff
@@ -116,19 +123,25 @@ case class CircuitState(
     * cycle all memories
     */
   def cycle(): Unit = {
+    vcdLoggerOption.foreach { vcd =>
+      vcd.incrementTime(10)
+      vcd.raiseClock
+    }
     registers.keys.foreach { key =>
       registers(key) = nextRegisters(key)
     }
 //    nextRegisters.clear()
 //    ephemera.clear()
     cycleMemories()
-    vcdLoggerOption.foreach { vcd =>
-      vcd.incrementTime()
-    }
     nameToConcreteValue = mutable.HashMap((inputPorts ++ outputPorts ++ registers).toSeq:_*)
     isStale = true
 
     stateCounter += 1
+
+    vcdLoggerOption.foreach { vcd =>
+      vcd.incrementTime(10)
```

Comment:
I'm working on patching up the scoping stuff a bit more this afternoon.  I'll think about the -1, I was using that a shorthand for initial state, but I can probably pick something more reasonable